vxquery-core/src/main/java/org/apache/vxquery/runtime/functions/json/LibjnDescendantObjectsScalarEvaluator.java

```diff
@@ -0,0 +1,127 @@
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+package org.apache.vxquery.runtime.functions.json;
+
+import org.apache.hyracks.algebricks.runtime.base.IScalarEvaluator;
+import org.apache.hyracks.api.context.IHyracksTaskContext;
+import org.apache.hyracks.data.std.api.IPointable;
+import org.apache.hyracks.data.std.primitive.UTF8StringPointable;
+import org.apache.hyracks.data.std.util.ArrayBackedValueStorage;
+import org.apache.vxquery.datamodel.accessors.SequencePointable;
+import org.apache.vxquery.datamodel.accessors.TaggedValuePointable;
+import org.apache.vxquery.datamodel.accessors.jsonitem.ArrayPointable;
+import org.apache.vxquery.datamodel.accessors.jsonitem.ObjectPointable;
+import org.apache.vxquery.datamodel.builders.sequence.SequenceBuilder;
+import org.apache.vxquery.datamodel.values.ValueTag;
+import org.apache.vxquery.datamodel.values.XDMConstants;
+import org.apache.vxquery.exceptions.ErrorCode;
+import org.apache.vxquery.exceptions.SystemException;
+import org.apache.vxquery.runtime.functions.base.AbstractTaggedValueArgumentScalarEvaluator;
+
+import java.io.IOException;
+
+public class LibjnDescendantObjectsScalarEvaluator extends AbstractTaggedValueArgumentScalarEvaluator {
+    protected final IHyracksTaskContext ctx;
+
+    private final SequencePointable sp, sp1;
+    private final UTF8StringPointable stringp;
+    private final SequenceBuilder sb;
+    private final ArrayBackedValueStorage abvs;
+
+    public LibjnDescendantObjectsScalarEvaluator(IHyracksTaskContext ctx, IScalarEvaluator[] args) {
+        super(args);
+        this.ctx = ctx;
+        sp = (SequencePointable) SequencePointable.FACTORY.createPointable();
+        sp1 = (SequencePointable) SequencePointable.FACTORY.createPointable();
+        stringp = (UTF8StringPointable) UTF8StringPointable.FACTORY.createPointable();
+        sb = new SequenceBuilder();
+        abvs = new ArrayBackedValueStorage();
+    }
+
+    @Override
+    protected void evaluate(TaggedValuePointable[] args, IPointable result) throws SystemException {
+        try {
+            abvs.reset();
+            sb.reset(abvs);
+            TaggedValuePointable tvp = args[0];
+            if (tvp.getTag() == ValueTag.SEQUENCE_TAG) {
+                TaggedValuePointable tempTvp = ppool.takeOne(TaggedValuePointable.class);
+                tvp.getValue(sp);
+                int size = sp.getEntryCount();
+                for (int i = 0; i < size; i++) {
+                    sp.getEntry(i, tempTvp);
+                    process(tempTvp);
+                }
+                ppool.giveBack(tempTvp);
+            } else if (tvp.getTag() == ValueTag.OBJECT_TAG) {
+                nestedObject(tvp);
+            } else if (tvp.getTag() == ValueTag.ARRAY_TAG) {
+                insideArray(tvp);
+            } else {
+                XDMConstants.setEmptySequence(tvp);
```

Comment:
The else statement does nothing. An empty sequence is returned because the sequence builder never gets a value. Please remove the else statement.